modules/bs.block/data/bs.block/function/produce/block_sound/play_block_sound.mcfunction

```diff
@@ -0,0 +1,29 @@
+# ------------------------------------------------------------------------------------------------------------
+# Copyright (c) 2025 Gunivers
+#
+# This file is part of the Bookshelf project (https://github.com/mcbookshelf/Bookshelf).
+#
+# This source code is subject to the terms of the Mozilla Public License, v. 2.0.
+# If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Conditions:
+# - You may use this file in compliance with the MPL v2.0
+# - Any modifications must be documented and disclosed under the same license
+#
+# For more details, refer to the MPL v2.0.
+# ------------------------------------------------------------------------------------------------------------
+
+data modify storage bs:ctx _ set value {}
```

Comment:
could be simplified to data modify storage bs:ctx _ set from storage bs:in block.play_block_sound

Reply:
Also looking at that, do we really need to pass the whole sounds map i feel like the user can do that better and it avoid the use of a macro, and actually also no need to log since the user is now responsible for checking if the storage contains the sound. Also it's as simple to use as your version:
```mcfunction
data modify storage bs:in block.play_block_sound set value { source: "block", targets: "@s", pos: "~ ~ ~", volume: 1, pitch: 1, min_volume: 0 }
data modify storage bs:in block.play_block_sound.sound set from storage bs:out block.sounds.break
```

Reply:
Indeed we can. Even if the idea was, like with other producing functions, to just be able to drop the whole block data inside the input.

Reply:
Yeah i can see the idea behind that but i'm not sure it's worth it considering that:
- the alternative is as easy to use
- has better performances
- is actually less code on our side (so less things to worry about)

Reply:
If you think this is better I can change it. But currently, I think we consider that users never have to know what is the structure of the virtual blocks. Here, we are breaking this expectation. Are you good with that?

Reply:
Personally i don't mind, also the fact that the user should not know what is the structure is not really true since we explain the format on the doc (what we dont explain is the extra key `_` that is used to manipulate the output). So the most important part is that it's readonly and that the user should not modify directly the output

Reply:
Also i think it'll be nice to add to the doc the different sound types that the module returns